internal/runners/push/push.go

```diff
@@ -64,28 +66,36 @@ func (r *Push) Run() *failures.Failure {
 		return nil
 	}
 
-	lang, langVersion, fail := r.languageForProject(r.project)
-	if fail != nil {
-		return fail
-	}
+	// try to create the project at the given commit id.
+	if r.project.CommitID() != "" {
+		err := model.CreateProjectAtCommit(r.project.Owner(), r.project.Name(), r.project.Private(), r.project.CommitUUID())
```

Comment:
Rather than having two specialized functions can we de-specialize the existing one. Same as before the logic here is pretty much identical. Just make a function that creates a project and another one that attaches a commit. This diff should be tiny.